cfspopcon/formulas/scrape_off_layer/lambda_q.py

```diff
@@ -42,58 +44,87 @@ def calc_lambda_q(
         :term:`lambda_q` [mm]
     """
     if lambda_q_scaling == LambdaQScaling.Brunner:
-        return lambda_q_factor * float(calc_lambda_q_with_brunner.unitless_func(average_total_pressure))
+        return float(
+            calc_lambda_q_with_brunner.unitless_func(average_total_pressure=average_total_pressure, lambda_q_factor=lambda_q_factor)
+        )
     elif lambda_q_scaling == LambdaQScaling.EichRegression14:
-        return lambda_q_factor * float(calc_lambda_q_with_eich_regression_14.unitless_func(B_pol_out_mid))
+        return float(calc_lambda_q_with_eich_regression_14.unitless_func(B_pol_out_mid=B_pol_out_mid, lambda_q_factor=lambda_q_factor))
     elif lambda_q_scaling == LambdaQScaling.EichRegression15:
-        return lambda_q_factor * float(
+        return float(
             calc_lambda_q_with_eich_regression_15.unitless_func(
-                power_crossing_separatrix, major_radius, B_pol_out_mid, inverse_aspect_ratio
+                power_crossing_separatrix=power_crossing_separatrix,
+                major_radius=major_radius,
+                B_pol_out_mid=B_pol_out_mid,
+                inverse_aspect_ratio=inverse_aspect_ratio,
+                lambda_q_factor=lambda_q_factor,
             )
         )
     else:
         raise NotImplementedError(f"No implementation for lambda_q scaling {lambda_q_scaling}")
 
 
+@Algorithm.register_algorithm(return_keys=["lambda_q"])
 @wraps_ufunc(
     return_units=dict(lambda_q=ureg.millimeter),
-    input_units=dict(average_total_pressure=ureg.atm),
+    input_units=dict(average_total_pressure=ureg.atm, lambda_q_factor=ureg.dimensionless),
 )
-def calc_lambda_q_with_brunner(average_total_pressure: float) -> float:
+def calc_lambda_q_with_brunner(average_total_pressure: float, lambda_q_factor: float = 1.0) -> float:
     """Return lambda_q according to the Brunner scaling.
 
     Equation 4 in :cite:`brunner_2018_heat_flux`
     """
-    return float(0.91 * average_total_pressure**-0.48)
+    return float(lambda_q_factor * 0.91 * average_total_pressure**-0.48)
+
+
+@Algorithm.register_algorithm(return_keys=["lambda_q"])
+@wraps_ufunc(
+    return_units=dict(lambda_q=ureg.millimeter),
+    input_units=dict(
+        magnetic_field_on_axis=ureg.T,
+        q_star=ureg.dimensionless,
+        power_crossing_separatrix=ureg.megawatt,
+        lambda_q_factor=ureg.dimensionless,
+    ),
+)
+def calc_lambda_q_with_eich_regression_9(
+    magnetic_field_on_axis: float, q_star: float, power_crossing_separatrix: float, lambda_q_factor: float = 1.0
+) -> float:
+    """Return lambda_q according to Eich regression 9.
+
+    #9 in Table 2 in :cite:`eich_scaling_2013`
+    """
+    return float(lambda_q_factor * 0.7 * magnetic_field_on_axis**-0.77 * q_star**1.05 * power_crossing_separatrix**0.09)
```

Comment:
Should this be a standalone alg or just be a branch in `calc_lambda_q` like the other options?

Reply:
Have added both. In the future I'd like to have all "calc_lambda_q_..." be standalone algorithms, instead of having a complicated set of inputs for "calc_lambda_q"